docs/audio/Soundfont_XML.md

```diff
@@ -0,0 +1,314 @@
+# Soundfont XML Format Specification
+
+Soundfont XMLs
+
+---
+
+```xml
+<Soundfont
+    Name="<C Identifier>"
+    Index="<uint>"
+    Medium="<Medium>"
+    CachePolicy="<CachePolicy>"
+    SampleBank="<Path>"
+    Indirect="[uint]"
+    SampleBankDD="[Path]"
+    IndirectDD="[uint]"
+    LoopsHaveFrames="[bool]"
+    PadToSize="[uint]"
+    NumInstruments="[uint]"
+>
+```
+Begins a new soundfont.
+
+**Properties**
+- **Name**: Soundfont symbol name. Must be a valid C identifier.
+- **Index**: Soundfont index. Must be an integer.
+- **Medium**: Storage medium. Must be an enum name from `SampleMedium`.
+- **CachePolicy**: Cache policy. Must be an enum name from `AudioCacheLoadType`.
+- **SampleBank**: Path to samplebank xml used by this soundfont.
+- <ins>[Optional]</ins> **Indirect**: Pointer index if the samplebank is referenced indirectly.
+- <ins>[Optional]</ins> **SampleBankDD**: Path to samplebank xml used for DD medium.
+- <ins>[Optional]</ins> **IndirectDD**: Pointer index if the DD samplebank is referenced indirectly.
+- <ins>[Optional]</ins> **LoopsHaveFrames**: Whether loops in this soundfont store the total frame count of the sample. Must be a boolean.
+- <ins>[Optional]</ins> **PadToSize**: For matching only. Specifies the total file size the result output should be padded to.
+- <ins>[Optional]</ins> **NumInstruments**: For matching only. Specifies the total number of instrument pointers. Usually this is automatically assigned based on `max(program_number) + 1` but some vanilla banks don't match this way.
+
+**Sub-Attributes**
+
+-
+    ```xml
+    <Envelopes>
+    ```
+    Lists envelopes defined in this soundfont.
+
+    **Properties**
+
+    N/A
+
+    **Sub-Attributes**
+
+    -
+        ```xml
+        <Envelope
+            Name="<C Identifier>"
+            Release="<u8>"
+        >
+        ```
+        Starts a new envelope.
+
+        **Properties**
+
+        - **Name**: Unique name for this envelope. Must be a valid C identifier.
+        - **Release**: Release rate index for this envelope
```

Comment:
what does this index into exactly? I tried searching for "release rate" but didn't find much

Reply:
It indexes `gAudioCtx.adsrDecayTable`, I mentioned this but didn't add much explanation for it, I think the code surrounding this probably needs some docs before reviewing again how it's presented here